src/parsers/add-font-styles.ts

```diff
@@ -0,0 +1,45 @@
+import { DeepKeyTokenMap, SingleToken } from '@tokens-studio/types';
+// @ts-expect-error no type exported for this function
+import getReferences from 'style-dictionary/lib/utils/references/getReferences.js';
+// @ts-expect-error no type exported for this function
+import usesReference from 'style-dictionary/lib/utils/references/usesReference.js';
+
+function recurse(
+  slice: DeepKeyTokenMap<false>,
+  boundGetRef: (ref: string) => Array<SingleToken<false>>,
+) {
+  for (const key in slice) {
+    const token = slice[key];
+    const { type } = token;
+    if (token.value && type) {
+      if (type === 'typography') {
+        if (typeof token.value.fontWeight === 'string' && usesReference(token.value.fontWeight)) {
+          let ref = { value: token.value.fontWeight };
+          while (ref && ref.value && typeof ref.value === 'string' && usesReference(ref.value)) {
+            try {
+              ref = Object.fromEntries(
+                Object.entries(boundGetRef(ref.value)[0]).map(([k, v]) => [k, v]),
+              );
+            } catch (e) {
+              console.warn(`Warning: could not resolve reference ${ref.value}`);
+              return;
+            }
+          }
+
+          if (ref.value.toLowerCase().indexOf('italic') > -1) {
+            token.value.fontStyle = 'italic';
```

Comment:
I'm guessing we should handle the `normal` and possibly `oblique` case here as well now that I think of it... 🤔

Reply:
Handled those two cases as well now I think 👍🏻